File: JAVA/level2/큰_수_만들기.java
class Solution {
    public String solution(String number, int k) {
        String answer = "";

        while(k != 0){
            if(number.length() == k)
                return answer;

            int maxVal = -1;
            int index = -1;

            for(int i=0; i<=k; i++){
                if(maxVal < number.charAt(i) - '0'){
                    maxVal = number.charAt(i) - '0';
                    index = i;
                }
            }
            answer += number.charAt(index);
            number = number.substring(index+1);
            k -= index;
        }

        if(number.length() != 0)
            answer += number;

        return answer;
    }
}

File: JavaScript/level2/큰_수_만들기.js
function solution(number, k) {
    var answer = '';
    let start = 0;
    let selected = [];
    let size = k;
    
    while(k > 0){
        let index = start;
        let maxVal = number[start];
            for(let i=start; i<Math.min(start+k+1, number.length); i++){
                if(maxVal === '9')
                    break;
                if(maxVal < number[i]){
                    maxVal = number[i];
                    index = i;
                }
            }
        selected.push(index);
        k -= (index - start);
        start = index+1;
        if(start+k >= number.length)
            break;
    }
    
    for(let i of selected)
        answer += number[i];

    return answer+number.substring(answer.length+size);
}